xds/internal/balancer/outlierdetection/balancer_test.go

```diff
@@ -270,20 +293,1097 @@ func (lbc *LBConfig) Equal(lbc2 *LBConfig) bool {
+	internal.RegisterOutlierDetectionBalancerForTesting()
+	builder := balancer.Get(Name)
+	if builder == nil {
+		t.Fatalf("balancer.Get(%q) returned nil", Name)
+	}
+	tcc := testutils.NewTestClientConn(t)
+	odB := builder.Build(tcc, balancer.BuildOptions{})
+	return odB.(*outlierDetectionBalancer), tcc, func() {
+		odB.Close()
+		internal.UnregisterOutlierDetectionBalancerForTesting()
+	}
+}
+
+type emptyChildConfig struct {
+	serviceconfig.LoadBalancingConfig
 }
 
-type errParseConfigBuilder struct{}
+// TestChildBasicOperations tests basic operations of the Outlier Detection
+// Balancer and it's interaction with it's child. The following scenarios are
+// tested, in a step by step fashion:
+// 1. The Outlier Detection Balancer receives it's first good configuration. The
+// balancer is expected to create a child and sent the child it's configuration.
+// 2. The Outlier Detection Balancer receives new configuration that specifies a
+// child's type, and the new type immediately reports READY inline. The first
+// child balancer should be closed and the second child balancer should receive
+// a config update.
+// 3. The Outlier Detection Balancer is closed. The second child balancer should
+// be closed.
+func (s) TestChildBasicOperations(t *testing.T) {
+	bc := emptyChildConfig{}
+
+	ccsCh := testutils.NewChannel()
+	closeCh := testutils.NewChannel()
+
+	stub.Register(t.Name()+"child1", stub.BalancerFuncs{
+		UpdateClientConnState: func(bd *stub.BalancerData, ccs balancer.ClientConnState) error {
+			ccsCh.Send(ccs.BalancerConfig)
+			return nil
+		},
+		Close: func(bd *stub.BalancerData) {
+			closeCh.Send(nil)
+		},
+	})
+
+	stub.Register(t.Name()+"child2", stub.BalancerFuncs{
+		UpdateClientConnState: func(bd *stub.BalancerData, _ balancer.ClientConnState) error {
+			// UpdateState inline to READY to complete graceful switch process
+			// synchronously from any UpdateClientConnState call.
+			bd.ClientConn.UpdateState(balancer.State{
+				ConnectivityState: connectivity.Ready,
+				Picker:            &testutils.TestConstPicker{},
+			})
+			ccsCh.Send(nil)
+			return nil
+		},
+		Close: func(bd *stub.BalancerData) {
+			closeCh.Send(nil)
+		},
+	})
+
+	od, tcc, _ := setup(t)
+	defer internal.UnregisterOutlierDetectionBalancerForTesting()
+
+	// This first config update should a child to be built and forwarded it's
+	// first update.
+	od.UpdateClientConnState(balancer.ClientConnState{
+		BalancerConfig: &LBConfig{
+			Interval:           10 * time.Second,
+			BaseEjectionTime:   30 * time.Second,
+			MaxEjectionTime:    300 * time.Second,
+			MaxEjectionPercent: 10,
+			SuccessRateEjection: &SuccessRateEjection{
+				StdevFactor:           1900,
+				EnforcementPercentage: 100,
+				MinimumHosts:          5,
+				RequestVolume:         100,
+			},
+			FailurePercentageEjection: &FailurePercentageEjection{
+				Threshold:             85,
+				EnforcementPercentage: 5,
+				MinimumHosts:          5,
+				RequestVolume:         50,
+			},
```

Comment:
You could probably reduce some lines in the test by making a variable for the configuration without the child policy config.

```golang
var baseCfgWithoutChildConfig = BalancerConfig: &LBConfig{
	Interval:           10 * time.Second,
	BaseEjectionTime:   30 * time.Second,
	MaxEjectionTime:    300 * time.Second,
	MaxEjectionPercent: 10,
	SuccessRateEjection: &SuccessRateEjection{
		StdevFactor:           1900,
		EnforcementPercentage: 100,
		MinimumHosts:          5,
		RequestVolume:         100,
	},
	FailurePercentageEjection: &FailurePercentageEjection{
		Threshold:             85,
		EnforcementPercentage: 5,
		MinimumHosts:          5,
		RequestVolume:         50,
	},
}
```

And change just the child policy config here and down below when calling `UpdateClientConnState`.